proposals/lambda-method-group-defaults.md

```diff
@@ -145,31 +190,44 @@ only been supported since C# 10, so only code which has been written since then
 ### Grammar and Parser Changes
 This enhancement requires the following changes to the grammar for lambda expressions.
 ```diff
-explicit_anonymous_function_parameter
+ explicit_anonymous_function_parameter_list
++    : explicit_anonymous_function_parameters
++    | explicit_anonymous_function_parameters ',' parameter_array
++    | parameter_array
++    ;
++
++explicit_anonymous_function_parameters
+     : explicit_anonymous_function_parameter
+       (',' explicit_anonymous_function_parameter)*
+     ;
+
+ explicit_anonymous_function_parameter
 -    : anonymous_function_parameter_modifier? type identifier
 +    : anonymous_function_parameter_modifier? type identifier default_argument?
-    ;
+     ;
 ```
 No changes to the grammar are necessary for method groups since this proposal would only change their semantics.
 
 ### Binder Changes
 
 #### Synthesizing New Delegate Types
-As with the behavior for delegates with `ref` or `out` parameters, a new natural type is generated for each lambda or method group defined with any default parameter values.
+As with the behavior for delegates with `ref` or `out` parameters, a new natural type is generated for each lambda or method group defined with any default or `params` parameters.
 Note that in the below examples, the notation `a'`, `b'`, etc. is used to represent these anonymous delegate types.
 
 ```csharp
 var addWithDefault = (int addTo = 2) => addTo + 1;
-// internal delegate int b'(int arg0 = 2);
+// internal delegate int a'(int arg0 = 2);
 var printString = (string toPrint = "defaultString") => Console.WriteLine(toPrint);
-// internal delegate void c'(string arg0 = "defaultString");
+// internal delegate void b'(string arg0 = "defaultString");
+var count = (params int[] xs) => xs.Length;
+// internal delegate int c'(params int[] arg0);
 string pathJoin(string s1, string s2, string sep = "/") { return $"{s1}{sep}{s2}"; }
 var joinFunc = pathJoin;
 // internal delegate string d'(string arg0, string arg1, string arg3 = " ");
 ```
 
 #### Conversion and Unification Behavior 
-The delegates described previously will be unified when the same parameter (based on position) has the same default value, regardless of parameter name.
+Anonymous delegates with default parameters will be unified when the same parameter (based on position) has the same default value, regardless of parameter name.
 
```

Comment:
Do we need a similar statement for delegates for `params` cases?

Reply:
Added: 8f426e696075d4da1bc724b7a52ababee9acf8c6